### PR TITLE
PGP interstitial

### PR DESCRIPTION
**Pull request by beaugunderson**

### CODE REVIEW

File: open_humans/urls.py [hunk 1]
```diff
@@ -86,6 +86,8 @@
         name='terms-of-use'),
     url(r'^activities/$', ActivitiesView.as_view(), name='activities'),
     url(r'^statistics/$', StatisticsView.as_view(), name='statistics'),
+    url(r'^pgp-interstitial/$', PGPInterstitialView.as_view(),
```
Comment (madprime) [inline open_humans/urls.py]: Let's call it 'pgp-quick-note'

File: open_humans/middleware.py [hunk 2]
```diff
@@ -78,3 +81,28 @@ def process_request(request):
...
+    def process_request(request):
+        if request.user.is_anonymous():
+            return
+
+        if request.user.member.seen_pgp_interstitial:
+            return
+
+        public = (request.user.pgp.datafile_set
+                  .filter(_public_data_access__is_public=True))
+        private = (request.user.pgp.datafile_set
```
Comment (beaugunderson) [inline open_humans/middleware.py]: use `is_enrolled` here instead

File: open_humans/middleware.py [hunk 3]
```diff
@@ -78,3 +81,28 @@ def process_request(request):
...
         return get_production_redirect(request)
+
+
+class PGPInterstitialRedirectMiddleware:
+    """
+    Redirect users with more than 1 private PGP datasets and zero public
+    datasets to an interstitial page exactly one time.
+    """
+    @staticmethod
+    def process_request(request):
```
Comment (madprime) [inline open_humans/middleware.py]: add return if not a PGP user

File: open_humans/middleware.py [hunk 4]
```diff
@@ -78,3 +81,28 @@ def process_request(request):
...
+
+        if request.user.member.seen_pgp_interstitial:
+            return
+
+        public = (request.user.pgp.datafile_set
+                  .filter(_public_data_access__is_public=True))
+        private = (request.user.pgp.datafile_set
+                   .filter(_public_data_access__is_public=False))
+
+        if private.count() > 0 and public.count() < 1:
```
Comment (madprime) [inline open_humans/middleware.py]: if public.count > 0 ... skip interstitial and change seen_pgp_interstitial to True

**Pull request merged**